build_tools/cmake/build_riscv.sh

```diff
@@ -55,16 +56,36 @@ else
   mkdir -p "${BUILD_RISCV_DIR?}"
 fi
 
-# Configure riscv for 64-bit linux config, then build.
-"${CMAKE_BIN?}" -G Ninja -B "${BUILD_RISCV_DIR?}" \
-  -DCMAKE_TOOLCHAIN_FILE="$(realpath ${ROOT_DIR?}/build_tools/cmake/riscv.toolchain.cmake)" \
-  -DIREE_HOST_BINARY_ROOT="$(realpath ${BUILD_HOST_DIR?}/install)" \
-  -DRISCV_CPU=rv64 \
-  -DIREE_BUILD_COMPILER=OFF \
-  -DIREE_ENABLE_MLIR=OFF \
-  -DIREE_BUILD_TESTS=ON \
-  -DIREE_BUILD_SAMPLES=OFF \
-  -DIREE_BUILD_EMBEDDING_SAMPLES=ON \
-  -DRISCV_TOOLCHAIN_ROOT="${RISCV_TOOLCHAIN_ROOT?}" \
-  "${ROOT_DIR?}"
+echo "Build riscv target with the config of ${RISCV_CONFIG?}"
+if [[ "${RISCV_CONFIG?}" == "rv64" ]]; then
+  # Configure riscv for 64-bit linux config, then build.
+  "${CMAKE_BIN?}" -G Ninja -B "${BUILD_RISCV_DIR?}" \
+    -DCMAKE_TOOLCHAIN_FILE="$(realpath ${ROOT_DIR?}/build_tools/cmake/riscv.toolchain.cmake)" \
+    -DIREE_HOST_BINARY_ROOT="$(realpath ${BUILD_HOST_DIR?}/install)" \
+    -DRISCV_CPU="${RISCV_CONFIG?}" \
+    -DIREE_BUILD_COMPILER=OFF \
+    -DIREE_ENABLE_MLIR=OFF \
+    -DIREE_BUILD_TESTS=ON \
+    -DIREE_BUILD_SAMPLES=OFF \
+    -DIREE_BUILD_EMBEDDING_SAMPLES=ON \
+    -DRISCV_TOOLCHAIN_ROOT="${RISCV_TOOLCHAIN_ROOT?}" \
+    "${ROOT_DIR?}"
+elif [[ "${RISCV_CONFIG?}" == "rv32-baremetal" ]]; then
+  # Configure riscv for 32-bit bare-metal config, then build.
+  # IREE_BUILD_TESTS is off until iree/tools are refactored to support threadless config.
+  "${CMAKE_BIN?}" -G Ninja -B "${BUILD_RISCV_DIR?}" \
+    -DCMAKE_TOOLCHAIN_FILE="$(realpath ${ROOT_DIR?}/build_tools/cmake/riscv.toolchain.cmake)" \
+    -DIREE_HOST_BINARY_ROOT="$(realpath ${BUILD_HOST_DIR?}/install)" \
+    -DRISCV_CPU="${RISCV_CONFIG?}" \
+    -DIREE_BUILD_COMPILER=OFF \
+    -DIREE_ENABLE_MLIR=OFF \
+    -DIREE_BUILD_TESTS=OFF \
+    -DIREE_BUILD_SAMPLES=OFF \
+    -DIREE_BUILD_EMBEDDING_SAMPLES=ON \
+    -DRISCV_TOOLCHAIN_ROOT="${RISCV_RV32_NEWLIB_TOOLCHAIN_ROOT?}" \
+    "${ROOT_DIR?}"
+else
+  echo "riscv config not supported yet"
+  return -1
+fi
```

Comment:
There's a lot of duplication in these two branches. Can you instead use an array for arguments and then only branch on the things that are different? Also has the advantage that you can put comments in there if you'd like and don't need to remember line continuations

Something like this:

```suggestion
declare -a args
args=(
  "-G" "Ninja"
  "-B" "${BUILD_RISCV_DIR?}"
  -DCMAKE_TOOLCHAIN_FILE="$(realpath ${ROOT_DIR?}/build_tools/cmake/riscv.toolchain.cmake)"
  -DIREE_HOST_BINARY_ROOT="$(realpath ${BUILD_HOST_DIR?}/install)"
  -DRISCV_CPU="${RISCV_CONFIG?}"
  -DIREE_BUILD_COMPILER=OFF
  -DIREE_ENABLE_MLIR=OFF
  -DIREE_BUILD_SAMPLES=OFF
  -DIREE_BUILD_EMBEDDING_SAMPLES=ON
)

if [[ "${RISCV_CONFIG?}" == "rv64" ]]; then
  args+=(
    -DIREE_BUILD_TESTS=ON
    -DRISCV_TOOLCHAIN_ROOT="${RISCV_TOOLCHAIN_ROOT?}"
  )
elif [[ "${RISCV_CONFIG?}" == "rv32-baremetal" ]]; then
  args+=(
    # Off until iree/tools are refactored to support threadless config.
    -DIREE_BUILD_TESTS=OFF
    -DRISCV_TOOLCHAIN_ROOT="${RISCV_RV32_NEWLIB_TOOLCHAIN_ROOT?}"
  )
else
  echo "riscv config not supported yet"
  return -1
fi
"${CMAKE_BIN?}" ${args} "${ROOT_DIR?}"
```

Reply:
Thanks for the suggestion. Refactor this part of the code.